src/pages/add/import-wallet/import-wallet.ts

```diff
@@ -85,19 +120,259 @@ export class ImportWalletPage implements OnInit {
     this.importForm.get('filePassword').updateValueAndValidity();
   }
 
-  setDerivationPath() {
-    this.formData.derivationPath = this.formData.testnet ? this.derivationPathForTestnet : this.derivationPathByDefault;
-  }
-
   normalizeMnemonic(words: string) {
     if (!words || !words.indexOf) return words;
     var isJA = words.indexOf('\u3000') > -1;
     var wordList = words.split(/[\u3000\s]+/);
 
     return wordList.join(isJA ? '\u3000' : ' ');
-  };
+  }
+
+  private processWalletInfo(code: string): void {
+    if (!code) return;
+
+    this.importErr = false;
+    let parsedCode = code.split('|');
+
+    if (parsedCode.length != 5) {
+      /// Trying to import a malformed wallet export QR code
+      this.popupProvider.ionicAlert('Error', 'Incorrect code format'); //TODO gettextcatalog
+      return;
+    }
+
+    let info = {
+      type: parsedCode[0],
+      data: parsedCode[1],
+      network: parsedCode[2],
+      derivationPath: parsedCode[3],
+      hasPassphrase: parsedCode[4] == 'true' ? true : false
+    };
+
+    if (info.type == '1' && info.hasPassphrase)
+      this.popupProvider.ionicAlert('Error', 'Password required. Make sure to enter your password in advanced options'); //TODO gettextcatalog
+
+    this.formData.derivationPath = info.derivationPath;
+    this.formData.testnetEnabled = info.network == 'testnet' ? true : false;
+    this.formData.words = info.data;
+  }
+
+  public switchTestnetOff(): void {
```

Comment:
Never used?

Reply:
Not at the moment, but we will use it in a future feature, when we add BCH

Reply:
Then should be added with BCH

Reply:
I removed it, thanks